src/geometry/CTiglIntersectionCalculation.cpp

```diff
@@ -136,6 +162,67 @@ CTiglIntersectionCalculation::CTiglIntersectionCalculation(CTiglShapeCache* cach
     computeIntersection(cache, hash1, hash2, shape, plane);
 }
 
+// Computes the intersection of a shape with a plane segment,
+// that is defined by two points p1 and p2 and a vector w in
+// the following way: p(u,v) = p1 (1-u) + p2u + wv, with
+// u in [0,1]. Ideally w should be perpendicular to (p1 - p2)
+// cache variable can be NULL
+CTiglIntersectionCalculation::CTiglIntersectionCalculation(CTiglShapeCache* cache,
+                                                           const std::string& shapeID,
+                                                           TopoDS_Shape shape,
+                                                           gp_Pnt point1,
+                                                           gp_Pnt point2,
+                                                           gp_Dir normal,
+                                                           bool forceOrthogonal)
+    : tolerance(1.0e-7)
+{
+
+    // create hash
+    size_t hash = boost::hash<std::string>()(shapeID);
+    size_t hashP1P2 = HashableProjection(point1,normal).HashValue() ^ HashableProjection(point2,normal).HashValue();
```

Comment:
I am still confused: The only two arguments against using xor as a hash combiner that I found online are the symmetry (`a xor b = b xor a`) (which we explicitly want) and the reason that (`a xor a = 0`). The latter case should be caught before it happens, before the call to `CTiglIntersectionCalculation.` We could also explicitly catch it in the `hash_combine_symmetric function`. Either way, it was not the problem on the Mac build.

Your suggestion for a combination function unfortunately resulted in a collision on a Windows x64 build (see commit 53977b0901c02955b8de6d7c89a91f98c88b212f) for the tiglIntersectWithPlane test.  I am now trying `(h1 ^ h2) ^ (h1 + h2)` but I am really grasping at straws...